iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -610,4 +610,39 @@ cxx::error<TimerError> Timer::createErrorFromErrno(const int32_t errnum) noexcep
 }
 
 } // namespace posix
+
+namespace clock
+{
+Timer::Timer(const iox::units::Duration timeToWait) noexcept
+    : m_measuringDuration(timeToWait)
+    , m_startTime(getCurrentMonotonicTime())
+{
+}
+
+bool Timer::hasTimerExpired() noexcept
+{
+    auto currentTime = getCurrentMonotonicTime();
+    auto elpasedTime = currentTime - m_startTime;
+
+    if (elpasedTime >= m_measuringDuration)
+    {
+        return true;
+    }
+    return false;
```

Comment:
Yes this looks good for me. I have added m_endTime which keeps the expiration time. So everytime when hasExpired() is called instead of subtraction the current time is evaluated w.r.t the m_endTime

On the remaining time i have introduced a new public function remainingTime() which returns the time duration until the timer is active. It returns 0 if the timer is already expired.